cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -388,10 +389,23 @@ func ValidateFeatureGates(featureGates map[string]bool, fldPath *field.Path) fie
 func ValidateAPIEndpoint(c *kubeadm.MasterConfiguration, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	endpoint, err := kubeadmutil.GetMasterEndpoint(c)
-	if err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath, endpoint, err.Error()))
+	// validates the ControlPlaneEndpoint, if defined (valid IP or DNS, eventually with a valid port)
+	if len(c.API.ControlPlaneEndpoint) > 0 {
```

Comment:
GetMasterHostPort also has this logic, we should probably unify - https://github.com/kubernetes/kubernetes/pull/62314/files